src/reaction/dual_porosity.cc

```diff
@@ -213,16 +206,32 @@ void DualPorosity::initialize_fields()
   //setting fields in data
   data_.set_mesh(*mesh_);
   
+  //initialization of DOF handler
+  static FE_P_disc<0> fe0(0);
+  static FE_P_disc<1> fe1(0);
+  static FE_P_disc<2> fe2(0);
+  static FE_P_disc<3> fe3(0);
+  shared_ptr<DOFHandlerMultiDim> dh = make_shared<DOFHandlerMultiDim>(*mesh_);
```

Comment:
Maybe use better initialization of DH.

Reply:
Create sngle common dof handler for reactions in OperatorSplitting constructor.
Implement ReactionTerm::set_dh(DH) and use it to pass the common DH to the top reaction term instance.
Modify all reaction subclasses to pass DH down through the reaction tree in their initialize() method.